FinniversKit/Sources/Components/KeyValueGridView/KeyValueGridView.swift

```diff
@@ -80,11 +80,41 @@ public class KeyValueGridView: UIView {
     }
 
     private func createCellView(for pair: KeyValuePair) -> UIView {
-        let stackView = UIStackView(axis: .vertical, spacing: Warp.Spacing.spacing25, alignment: .leading, distribution: .equalSpacing, withAutoLayout: true)
+        let stackView = UIStackView(
+            axis: .vertical,
+            spacing: Warp.Spacing.spacing25,
+            alignment: .leading,
+            distribution: .equalSpacing,
+            withAutoLayout: true
+        )
 
         let titleLabel = Label(style: titleStyle, numberOfLines: 2, withAutoLayout: true)
         titleLabel.lineBreakMode = .byWordWrapping
 
+        let titleContainer = UIStackView()
+        titleContainer.axis = .horizontal
+        titleContainer.alignment = .center
+        titleContainer.distribution = .fill
+        titleContainer.spacing = Warp.Spacing.spacing100
+
+        titleLabel.text = pair.title
+
+        titleContainer.addArrangedSubview(titleLabel)
+        if let infoText = pair.infoTooltip, !infoText.isEmpty {
+            let infoButton = UIButton(type: .custom)
+            infoButton.setImage(Warp.Icon.info.uiImage, for: .normal)
+            infoButton.translatesAutoresizingMaskIntoConstraints = false
+            NSLayoutConstraint.activate([
+                infoButton.widthAnchor.constraint(equalToConstant: Warp.Spacing.spacing200),
+                infoButton.heightAnchor.constraint(equalToConstant: Warp.Spacing.spacing200)
+            ])
+            infoButton.addAction(UIAction(handler: { [weak self] _ in
+                self?.showTooltip(infoText, from: infoButton)
+            }), for: .touchUpInside)
```

Comment:
This will create a strong reference to `infoButton` from the closure. We should either weakify it, or use `action.sender` instead.

```swift
infoButton.addAction(UIAction(handler: { [weak self] action in
    guard let view = action.sender as? UIView else { return }
    self?.showTooltip(infoText, from: view)
}), for: .touchUpInside)
```